### PR TITLE
Challenge implementation

### PR DESCRIPTION
**Pull request by JErazo7**
Attached is the implementation of the challenge, I look forward to your prompt review! 

Thank you very much

### CODE REVIEW

File: lib/data/failures/restaurant_failure.dart [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'restaurant_failure.freezed.dart';
+
+@freezed
+class RestaurantFailure with _$RestaurantFailure {
+  const factory RestaurantFailure.unexpected() = Unexpected;
```
Comment (jshaw-mgm) [inline lib/data/failures/restaurant_failure.dart:7]: question: I only see a test for the unauthorized failure.  Do we need the other 2 to be tested?

File: lib/data/repositories/yelp_repository.dart [hunk 2]
```diff
@@ -58,22 +48,37 @@ class YelpRepository {
...
+        case 401:
+          return left(const RestaurantFailure.unauthorized());
+        case 500:
+          return left(const RestaurantFailure.serverError());
+        default:
+          return left(const RestaurantFailure.unexpected());
+      }
     } catch (e) {
-      return null;
+      return left(const RestaurantFailure.unexpected());
```
Comment (jshaw-mgm) [inline lib/data/repositories/yelp_repository.dart:74]: kudos: there may be cases in more complex apps where we want to deal with more response cases but I appreciate that you default to `unexpected` for this and seems like the appropriate amount of checks.  Nicely done.

File: lib/ui/router/router.dart [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import 'package:equatable/equatable.dart';
+import 'package:go_router/go_router.dart';
+import 'package:restaurantour/ui/models/restaurant_ui.dart';
+import 'package:restaurantour/ui/views/views.dart';
+
+class RouteName extends Equatable {
+  const RouteName._();
```
Comment (jshaw-mgm) [inline lib/ui/router/router.dart:7]: kudos: I really like the use of private constructors.

**Review by jshaw-mgm — COMMENTED** [review]
Thank you so much for submitting this PR so quickly.  My apologies for the delayed review.  

This code looks really awesome and I didn't have a lot of feedback because I think you did such an incredible job.  The app works really well, matches the designs and is well tested.

File: lib/data/failures/restaurant_failure.dart (hunk 1 above)
Comment (JErazo7) [reply lib/data/failures/restaurant_failure.dart:7]: You are right, I would include them when implementing unit tests for the repository as such. On the view side they are also useful to check the different messages that could be displayed according to the error.

**Comment by JErazo7** [comment]
> Thank you so much for submitting this PR so quickly. My apologies for the delayed review.
> 
> This code looks really awesome and I didn't have a lot of feedback because I think you did such an incredible job. The app works really well, matches the designs and is well tested.

I am very excited about your comments. Thank you very much for reviewing my code. I'm glad to know that my knowledge can be up to par and appreciated by people with an amazing track record like yours.

**Pull request closed without merging**